modules/core/test/test_misc.cpp

```diff
@@ -917,5 +917,45 @@ REGISTER_TYPED_TEST_CASE_P(Rect_Test, Overflows);
 typedef ::testing::Types<int, float, double> RectTypes;
 INSTANTIATE_TYPED_TEST_CASE_P(Negative_Test, Rect_Test, RectTypes);
 
+// Expected that SkipTestException thrown in the constructor should skip test but not fail
+struct TestFixtureSkip: public ::testing::Test {
+    TestFixtureSkip(bool throwEx = true) {
+        if (throwEx) {
+            throw SkipTestException("Skip test at constructor");
+        }
+    }
+};
+
+TEST_F(TestFixtureSkip, NoBodyRun) {
+    FAIL() << "Unreachable code called";
+}
+
+struct TestSetUpSkip: public ::testing::Test {
+    virtual void SetUp() {
+        throw SkipTestException("Skip test at SetUp");
+    }
+};
+
+TEST_F(TestSetUpSkip, NoBodyRun) {
+    FAIL() << "Unreachable code called";
+}
+
+// Check that single test skip won't break others
+TEST(TestCaseSkip, skipped_test)
+{
+    throw SkipTestException("Skip test");
+}
+
+static bool testRun = false;
+TEST(TestCaseSkip, executed_test)
+{
+    testRun = true;
+}
+
+TEST(TestCaseSkipCheck, run)
+{
+    ASSERT_TRUE(testRun);
+}
```

Comment:
The solution makes the 2 test dependent. In case if GTest shuffles them or one of them is disabled the result will be unexpected. Why do you want to do this?

Reply:
Indeed. Removed the test.